include/seqan3/argument_parser/detail/help_format.hpp

```diff
@@ -0,0 +1,668 @@
+#include <seqan3/core/detail/terminal.hpp>
+#include <seqan3/version.hpp>
+
+namespace seqan3::detail
+{
+//!\privatesection
+
+/*!\brief The format that prints the help page to std::cout.
+ * \ingroup argument_parser
+ *
+ * \details
+ * The help page printing is not done immediatly, because the user might not
+ * provide meta information, positional options, etc. in the correct order.
+ * In addition the needed order would be different from the parse format.
+ * Thus the calls are stored (parser_set_up_calls and positional_option_calls)
+ * and only evaluated when calling help_format::parse().
+ *
+ */
+class help_format
+{
+public:
+    /*!\brief The constructor
+     *
+     * \param advanced Whether to show advanced options or not.
+     */
+    help_format(bool advanced):
+        show_advanced_options(advanced)
+    {}
+
+    /*!\name Constructors
+     * \brief The (default) constructors 
+     * \{
+     */
+    help_format() = delete;
+    help_format(help_format const & pf) = default;
+    help_format & operator=(help_format const & pf) = default;
+    help_format(help_format &&) = default;
+    help_format & operator=(help_format &&) = default;
+    //!\}
+
+    /*!\brief Adds a print_list_item call to be evaluated later on.
+     *
+     * \param value The variable in which to store the given command line argument.
+     * \param short_id The short identifier for the option (e.g. `'i'`).
+     * \param long_id The long identifier for the option (e.g. `"integer"`).
+     * \param desc The description of the option.
+     * \param spec Advanced option specification. see seqan3::option_spec.
+     * \param validator The validator applied to the value after parsing (callable).
+     *
+     */
+    template <typename option_t, typename validator_type>
+    void add_option(option_t & value,
+                    char const short_id,
+                    std::string const & long_id,
+                    std::string const & desc,
+                    option_spec const & spec,
+                    validator_type && validator)
+    {
+        parser_set_up_calls.push_back([=, &value]()
+        {
+            if (!(spec & option_spec::HIDDEN) && (!(spec & option_spec::ADVANCED) || show_advanced_options))
+                print_list_item(_prep_id_for_help(short_id, long_id) + " " + _option_type_and_list_info(value),
+                                 (desc + " " + validator.get_help_page_message()));
+        });
+    }
+
+    /*!\brief Adds a print_list_item call to be evaluated later on.
+     *
+     * \param value The variable in which to store the given command line argument.
+     * \param short_id The short identifier for the flag (e.g. `'i'`).
+     * \param long_id The long identifier for the flag (e.g. `"integer"`).
+     * \param desc The description of the flag (which is only needed for the interface here).
+     * \param spec Advanced flag specification. see seqan3::option_spec.
+     *
+     */
```

Comment:
not quite. see line 121 `(which is only needed for the interface here).`
but i guess that's a neglectable hint.